Watchman.DomainModel/Configuration/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Autofac;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Configuration.Services
+{
+    public class ConfigurationService : IConfigurationService
+    {
+        private const ulong DEFAULT_SERVER_ID = 0;
+        private readonly IComponentContext _componentContext;
+        private readonly ISessionFactory _sessionFactory;
+        private readonly ConfigurationMapperService _configurationMapperService;
+        private readonly ConfigurationItemsSearcherService _configurationTypesSearcher;
+        private static Dictionary<Guid, int> _configurationVersions;
+        private static Dictionary<Type, Dictionary<ulong, IMappedConfiguration>> _cachedConfigurationItem;
+
+        public ConfigurationService(IComponentContext componentContext, ISessionFactory sessionFactory, ConfigurationMapperService configurationMapperService, ConfigurationItemsSearcherService configurationTypesSearcher)
+        {
+            this._componentContext = componentContext;
+            this._sessionFactory = sessionFactory;
+            this._configurationMapperService = configurationMapperService;
+            this._configurationTypesSearcher = configurationTypesSearcher;
+            this.Refresh();
```

Comment:
tak teoretycznie to stworzenie obiektu nie powinno nieść za sobą skutków ubocznych 
dlatego konstruktory nie powinny zawierać logiki

Reply:
ale praktycznie - hmm, tak chyba będzie bardziej użytecznie